src/main/java/org/opentripplanner/routing/impl/ATLFareServiceImpl.java

```diff
@@ -0,0 +1,378 @@
+package org.opentripplanner.routing.impl;
+
+import com.google.common.collect.Lists;
+import org.opentripplanner.model.Route;
+import org.opentripplanner.routing.core.Fare;
+import org.opentripplanner.routing.core.FareRuleSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.*;
+
+public class ATLFareServiceImpl extends DefaultFareServiceImpl {
+    public static final String COBB_AGENCY_ID = "2";
+    public static final String XPRESS_AGENCY_ID = "6";
+    public static final String MARTA_AGENCY_ID = "5";
+    public static final String GCT_AGENCY_ID = "4";
+
+    private enum TransferType {
+        END_TRANSFER, // Ends this transfer entirely.
+        NO_TRANSFER, // Effectively no transfer, but don't invalidate this transfer
+        FREE_TRANSFER, // Transfer is free
+        TRANSFER_WITH_UPCHARGE, // Transfer has a set upcharge
+        TRANSFER_PAY_DIFFERENCE // Transfer pays difference between default fares
+    }
+
+    private enum RideType {
+        FREE_RIDE,
+        MARTA,
+        COBB_LOCAL, COBB_EXPRESS("100", "101", "102"),
+        GCT_LOCAL, GCT_EXPRESS_Z1("102", "103a", "110", "swpr"), GCT_EXPRESS_Z2("101", "103"),
+        XPRESS_MORNING, XPRESS_AFTERNOON,
+        STREETCAR("atlsc");
+        private final HashSet<String> routeNames;
+        RideType(String... members){
+            this.routeNames = new HashSet<String>(Arrays.asList(members));
+        }
```

Comment:
```suggestion
        private final Set<String> routeNames;
        RideType(String... members){
            this.routeNames = Arrays.stream(members).map(String::toLowerCase).collect(Collectors.toSet());
        }
```

Reply:
Thank you! I wasn't sure how to handle all these types. This was my first time really writing Java in a looong time.